### PR TITLE
build: update to nodejs rules 0.34.0 and bazel 0.28.1

### PR DESCRIPTION
**Pull request by gregmagolan**
nodejs rules 0.34.0 now includes protractor_web_test_suite rule (via new @bazel/protractor rule) so we switch to that location for that rule in this PR so that /packages/bazel/src/protractor can be removed in a future PR

this PR also brings in node toolchain support which was released in nodejs rules 0.33.0. this is a prerequisite for RBE for mac & windows users

bazel schematics also updated with the same. @bazel/bazel 0.28.1 npm package includes transitive dep on hide-bazel-files so we're able to remove an explicit dep on that as well.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by gregmagolan** [comment]
Flakesobserved in CircleCI `test_docs_examples` & `test_docs_example_ivy` jobs:
```
============== AIO example output for: /home/circleci/ng/aio/content/examples/testing/

running: yarn e2e --prod --no-webdriver-update --port=4002
$ ng e2e --prod --no-webdriver-update --port=4002
****************************************************************************************
This is a simple server for use in testing or debugging Angular applications locally.
It hasn't been reviewed for security issues.

DON'T USE IT FOR PRODUCTION!
****************************************************************************************

Date: 2019-07-24T19:45:33.008Z
Hash: b838c82053d05f12bad3
Time: 51182ms
chunk {0} runtime.9788aa6691f1dd7e936d.js, runtime.9788aa6691f1dd7e936d.js.map (runtime) 2.2 kB [entry] [rendered]
chunk {1} 1.8b3441fcb21fbf004780.js, 1.8b3441fcb21fbf004780.js.map () 5.06 kB  [rendered]
chunk {2} main.e0eb13d9ee36ee450950.js, main.e0eb13d9ee36ee450950.js.map (main) 520 kB [initial] [rendered]
chunk {3} polyfills.b09e96b60ba8ebe82931.js, polyfills.b09e96b60ba8ebe82931.js.map (polyfills) 61.9 kB [initial] [rendered]
chunk {4} polyfills-es5.5183b5a01abde18047d8.js, polyfills-es5.5183b5a01abde18047d8.js.map (polyfills-es5) 68.1 kB [initial] [rendered]
chunk {5} styles.3d83f887483de9a3e746.css, styles.3d83f887483de9a3e746.css.map (styles) 20.6 kB [initial] [rendered]
** Angular Live Development Server is listening on localhost:4002, open your browser on http://localhost:4002/ **
ℹ ｢wdm｣: Compiled successfully.
[19:45:33] I/launcher - Running 1 instances of WebDriver
[19:45:33] I/direct - Using ChromeDriver directly...
[19:45:33] E/launcher - unknown error: Chrome failed to start: exited abnormally
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=75.0.3770.90 (a6dcaf7e3ec6f70a194cc25e8149475c6590e025-refs/branch-heads/3770@{#1003}),platform=Linux 4.15.0-1035-aws x86_64)
[19:45:33] E/launcher - WebDriverError: unknown error: Chrome failed to start: exited abnormally
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=75.0.3770.90 (a6dcaf7e3ec6f70a194cc25e8149475c6590e025-refs/branch-heads/3770@{#1003}),platform=Linux 4.15.0-1035-aws x86_64)
    at Object.checkLegacyResponse (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/error.js:546:15)
    at parseHttpResponse (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/http.js:509:13)
    at doSend.then.response (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/http.js:441:30)
    at process._tickCallback (internal/process/next_tick.js:68:7)
From: Task: WebDriver.createSession()
    at Function.createSession (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/webdriver.js:769:24)
    at Function.createSession (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/chrome.js:761:15)
    at Direct.getNewDriver (/home/circleci/ng/aio/tools/examples/shared/node_modules/protractor/built/driverProviders/direct.js:77:33)
    at Runner.createBrowser (/home/circleci/ng/aio/tools/examples/shared/node_modules/protractor/built/runner.js:195:43)
    at q.then.then (/home/circleci/ng/aio/tools/examples/shared/node_modules/protractor/built/runner.js:339:29)
    at _fulfilled (/home/circleci/ng/aio/tools/examples/shared/node_modules/q/q.js:834:54)
    at /home/circleci/ng/aio/tools/examples/shared/node_modules/q/q.js:863:30
    at Promise.promise.promiseDispatch (/home/circleci/ng/aio/tools/examples/shared/node_modules/q/q.js:796:13)
    at /home/circleci/ng/aio/tools/examples/shared/node_modules/q/q.js:556:49
    at runSingle (/home/circleci/ng/aio/tools/examples/shared/node_modules/q/q.js:137:13)
    at flush (/home/circleci/ng/aio/tools/examples/shared/node_modules/q/q.js:125:13)
    at process._tickCallback (internal/process/next_tick.js:61:11)
[19:45:33] E/launcher - Process exited with error code 199
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
completed: yarn e2e --prod --no-webdriver-update --port=4002
```

File: packages/compiler-cli/integrationtest/test.js [hunk 1]
```diff
@@ -11,9 +11,23 @@ const shx = require('shelljs');
 const os = require('os');
 const {runCommand, setupTestDirectory} = require('./test_helpers');
 
+function nodejs_workspace() {
```
Comment (alexeagle) [inline packages/compiler-cli/integrationtest/test.js]: this could use some commenting

File: packages/core/test/strict_types/BUILD.bazel [hunk 2]
```diff
@@ -7,7 +7,7 @@ ts_config(
     name = "tsconfig",
     src = "tsconfig.json",
     deps = [
-        "//tools:tsconfig-test",
+        "//packages:tsconfig-test",
```
Comment (alexeagle) [inline packages/core/test/strict_types/BUILD.bazel]: why?

**Review by alexeagle — APPROVED** [review]

File: packages/core/test/strict_types/BUILD.bazel (hunk 2 above)
Comment (gregmagolan) [reply packages/core/test/strict_types/BUILD.bazel]: Fixes build failure as `/packages/tsconfig-test.js` is the tsconfig that `packages/core/test/strict_types/tsconfig.json` extends from: 

```
{
  "extends": "../../../tsconfig-test.json",
  "compilerOptions": {
    "strict": true
  }
}
```

I don't why it was working with `//tools:tsconfig-test` prior to this.

File: packages/compiler-cli/integrationtest/test.js (hunk 1 above)
Comment (gregmagolan) [reply packages/compiler-cli/integrationtest/test.js]: 👍

**Review by devversion — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]
thanks!

**Comment by IgorMinar** [comment]
http://test/OCL:260045048:BASE:260045068:1564097631205:f15455e4

**Comment by IgorMinar** [comment]
merge-assistance: global approval

**Pull request closed without merging**